### DISCUSSION TITLE
Can't access error property on safeParse() result?

### DISCUSSION THREAD

**Discussion by nemanjam** [colinhacks/zod#1080] [Q&A]
I have type check that success is false but still can't access error property. When I log the result I see that error property exists.

```ts
const validationResult = postsGetSchema.safeParse(query);
if (!validationResult.success) {
  throw ApiError.ZodErrorToApiError(validationResult.error);
```

Example from Readme:

```ts
const result = stringSchema.safeParse("billie");
if (!result.success) {
  // handle error then return
  result.error;
} else {
  // do something
  result.data;
}
```

![image](https://user-images.githubusercontent.com/9990165/162792782-bf2fb3de-7cea-4cb4-9c02-7123aa02fbae.png)

**Comment by sylvestertarnowski** [comment]
There was an update in Typescript 4.4 that improved type guard narrowing, maybe bumping typescript version to this or above could help?
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html#control-flow-analysis-of-aliased-conditions-and-discriminants

**Reply by nemanjam** [reply]
It works when I set `"strict": true ` in `tsconfig.json`.

**Reply by sylvestertarnowski** [reply]
I assume it enabled Strict Null Checks, which would allow type narrowing to know when error is undefined, and when it is defined:
https://www.typescriptlang.org/tsconfig#strictNullChecks

**Reply by nemanjam** [reply]
Zod Readme says strict mode must be enabled.

https://github.com/colinhacks/zod#installation
